client/telemetry/src/lib.rs

```diff
@@ -333,61 +372,90 @@ impl TelemetryHandle {
 	///
 	/// The `connection_message` argument is a JSON object that is sent every time the connection
 	/// (re-)establishes.
-	pub fn start_telemetry(
-		&mut self,
-		span: TelemetrySpan,
-		endpoints: TelemetryEndpoints,
-		connection_message: ConnectionMessage,
-	) -> TelemetryConnectionNotifier {
-		let Self { message_sender } = self;
+	pub fn start_telemetry(&mut self, connection_message: ConnectionMessage) {
+		let Self {
+			message_sender: _,
+			register_sender,
+			id,
+			connection_notifier: _,
+			endpoints,
+		} = self;
 
-		let connection_notifier = TelemetryConnectionNotifier {
-			message_sender: message_sender.clone(),
-			addresses: endpoints.0.iter().map(|(addr, _)| addr.clone()).collect(),
+		let endpoints = match endpoints.take() {
+			Some(x) => x,
+			None => {
+				error!(
+					target: "telemetry",
+					"This telemetry instance has already been initialized!",
+				);
+				return;
+			}
 		};
 
-		match span.0.id() {
-			Some(id) => {
-				match message_sender.unbounded_send(Register::Telemetry {
-					id,
-					endpoints,
-					connection_message,
-				}) {
-					Ok(()) => {}
-					Err(err) => error!(
-						target: "telemetry",
-						"Could not initialize telemetry: \
-						the telemetry is probably already running: {}",
-						err,
-					),
-				}
-			}
-			None => error!(
+		match register_sender.unbounded_send(Register::Telemetry {
+			id: *id,
+			endpoints,
+			connection_message,
+		}) {
+			Ok(()) => {}
+			Err(err) => error!(
 				target: "telemetry",
-				"Could not initialize telemetry: the span could not be entered",
+				"Could not initialize telemetry: \
```

Comment:
Please return an error instead of logging something.

Reply:
Done in 6dd32f4664